src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XDocument.cs

```diff
@@ -112,7 +112,7 @@ public XDocument(XDocument other)
         /// <summary>
         /// Gets the XML declaration for this document.
         /// </summary>
-        public XDeclaration Declaration
+        public XDeclaration? Declaration
```

Comment:
constructors taking declaration should take nullable declaration

Reply:
What's the reason for that? Is it because there is no condition that throws for null?

Reply:
yes, declaration is optional in XML documents so I don't believe it should be required in the APIs